src/io/uart.rs

```diff
@@ -21,7 +21,7 @@ impl<T: Instance> UartMarker for Uart<'_, T> {}
 fn UART0() {
     let uart = unsafe { &*UART0::ptr() };
 
-    while uart.status.read().rxfifo_cnt().bits() > 0 {
+    while uart.status().read().rxfifo_cnt().bits() > 0 {
         let offset = if cfg!(feature = "esp32s2") {
             0x20C0_0000
```

Comment:
problem with S2 here is that below we offset  `*u32` but this offset is in bytes - just adding `/ 4` should make things work on ESP32-S2, too